File: Assignment/Abstraction/Statements/IfStatementNode.cs
using System.Collections.Generic;

namespace Assignment.Abstraction.Statements
{
    internal struct IfStatementNode : IBooleanStatementNode
    {
        public IfStatementNode(IEnumerable<INode> body, INode expression, INode child)
        {
            this.Body = body;
            this.Expression = expression;
            this.Child = child;
        }

        public IEnumerable<INode> Body { get; }
        public INode Expression { get; }
        public INode Child { get; }

        public string DisplayName()
        {
            return nameof(IfStatementNode);
        }
    }
}

File: src/main/java/ce305/abstraction/statements/IfStatementNode.java
package ce305.abstraction.statements;

import java.util.List;
import java.util.UUID;

import ce305.abstraction.INode;

public class IfStatementNode extends INode {
    public final List<INode> body;
    public final INode child;
    public final INode expression;

    public IfStatementNode(List<INode> body, INode expression, INode child) {
        super();
        this.body = body;
        this.child = child;
        this.expression = expression;
    }

    public IfStatementNode(List<INode> body, INode expression, INode child, UUID id) {
        super(id);
        this.body = body;
        this.child = child;
        this.expression = expression;
    }

    @Override
    public String getDisplayName() {
        return IfStatementNode.class.getSimpleName();
    }
}